lib/tree.rb

```diff
@@ -1,25 +1,46 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @height=0
+    @age=0
+    @apples=[]
+    @alive=true
   end
 
   def age!
+    @age += 1
+    if @age > 50
+      @alive=false
+      return
+    end
+
+    @height += 1
+    self.add_apples
   end
 
   def add_apples
+    num_apples = rand(1..5) * @age
+    i=0
+    while i < num_apples
```

Comment:
Check out the [`times` method](http://ruby-doc.org/core-2.4.2/Integer.html#method-i-times) to simplify this.